### PR TITLE
Hot fix access invalid memory

### PR DESCRIPTION
**Pull request by 0xakk0r0kamui**
- Check nil Output coin ws_account.go
- Check layer and role before publish node state (a.k.a peer state)

### CODE REVIEW

File: blockchain/synker.go [hunk 1]
```diff
@@ -208,6 +211,23 @@ func (synker *Synker) Start() {
...
 
+func (synker *Synker) checkStateAndPublishState() error {
+	// serverObj := synker.blockchain.config.ConsensusEngine.GetCurrentMiningPublicKey()
+	engine := synker.blockchain.config.ConsensusEngine
+	userKey, _ := engine.GetCurrentMiningPublicKey()
+	if userKey == "" {
+		return errors.New("Can not load current mining key")
+	}
+	userLayer, userRole, shardID := engine.GetUserRole()
+	if (userRole == common.CommitteeRole) || (userRole != common.ValidatorRole) || (userRole != common.ProposerRole) {
```
Comment (0xbunyip) [inline blockchain/synker.go]: GetUserRole returns userRole = CommitteeRole/PendingRole/WaitingRole only, check for ValidatorRole and ProposerRole is redundant (pls confirm with @0xBahamoot )

File: blockchain/blockchain.go [hunk 2]
```diff
@@ -68,7 +68,7 @@ type Config struct {
 	RandomClient      btc.RandomClient
 	Server            interface {
 		BoardcastNodeState() error
-		PublishNodeState() error
+		PublishNodeState(userLayer string, shardID int) error
```
Comment (0xbunyip) [inline blockchain/blockchain.go]:
```suggestion
		PublishNodeState() error
```

Comment (0xakk0r0kamui) [reply blockchain/blockchain.go]: My bad. Added missing file in new commit

File: blockchain/synker.go (hunk 1 above)
Comment (0xakk0r0kamui) [reply blockchain/synker.go]: Checked and removed ValidatorRole and ProposerRole

**Pull request merged**